vispy/visuals/shaders/compiler.py

```diff
@@ -36,7 +37,7 @@ class Compiler(object):
     def __init__(self, **shaders):
         # cache of compilation results for each function and variable
         self._object_names = {}  # {object: name}
-        self.shaders = shaders
+        self.shaders = weakref.WeakValueDictionary(**shaders)
```

Comment:
It is necessary because this creates a circular reference otherwise:

https://github.com/vispy/vispy/pull/729/files#diff-08d5869152f09fef8a2a5f3ed4a9261bL58